nbcelltests/tests/test_test.py

```diff
@@ -5,9 +5,223 @@
 # This file is part of the nbcelltests library, distributed under the terms of
 # the Apache License 2.0.  The full license can be found in the LICENSE file.
 #
-# for Coverage
+import tempfile
+import os
+import sys
+import unittest
 
+from nbcelltests.test import run
 
-class TestTests:
-    def test_run(self):
+# TODO: we should generate the notebooks rather than having them as files
+# (same for lint ones)
+CUMULATIVE_RUN = os.path.join(os.path.dirname(__file__), '_cumulative_run.ipynb')
+CELL_ERROR = os.path.join(os.path.dirname(__file__), '_cell_error.ipynb')
+TEST_ERROR = os.path.join(os.path.dirname(__file__), '_test_error.ipynb')
+TEST_FAIL = os.path.join(os.path.dirname(__file__), '_test_fail.ipynb')
+
+# Hack. We want to test expected behavior in distributed situation,
+# which we are doing via pytest --forked.
+FORKED = '--forked' in sys.argv
+
+
+def _assert_x_undefined(t):
+    t.run_test("""
+    try:
+        x
+    except NameError:
         pass
+    else:
+        raise Exception('x was already defined')
+    """)
+
+# TODO: This test file's manual use of unittest is brittle
+
+
+def _import_from_path(pypath, f):
+    import importlib.util
+    spec = importlib.util.spec_from_file_location(pypath, f)
+    mod = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(mod)
+    return mod
+
+
+class _TestCellTests(unittest.TestCase):
+    # abstract :)
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Generate test file from notebook, then import it, and make the
+        resulting module available as "generated_tests" class attribute.
+        """
+        tf = tempfile.NamedTemporaryFile(mode='w', suffix='.py', delete=False, encoding='utf8')
+        tf_name = tf.name
+        try:
+            cls.generated_tests = _import_from_path("nbcelltests.tests.%s.%s" % (__name__, cls.__name__), run(cls.NBNAME, filename=tf_name))
```

Comment:
This is a pretty obtuse line of code. Some comments would be nice :) E.g.:
- Why is the class name included in the import location spec? Is this merely for display purposes, or does it require class names to follow some formula?
- Maybe write a little about what `generated_tests` will be after this is called, and how it should be used.

Reply:
Agree with your first bullet point (and that the line is obtuse).

For the second bullet point, is the existing method docstring enough?